src/GitHub.App/Services/GitHubCredentialProvider.cs

```diff
@@ -34,21 +34,21 @@ public Credentials HandleCredentials(string url, string username, SupportedCrede
                 return null; // wondering if we should return DefaultCredentials instead
 
             var host = HostAddress.Create(url);
-            return secureCache.GetObject<Tuple<string, SecureString>>("login:" + host.CredentialCacheKeyHost)
-                .Select(CreateCredentials)
-                .Catch(Observable.Return<Credentials>(null))
-                .Wait();
-        }
-
-        static Credentials CreateCredentials(Tuple<string, SecureString> data)
-        {
-            Guard.ArgumentNotNull(data, nameof(data));
 
-            return new SecureUsernamePasswordCredentials
+            try
+            {
+                var credentials = ThreadHelper.JoinableTaskFactory.Run(async () => await keychain.Load(host));
```

Comment:
Previously we were doing a `.Wait` on an observable here, which is Bad. I've changed it to use the pattern described in 
[the 3 Threading Rules of VS](https://github.com/Microsoft/vs-threading/blob/master/doc/threading_rules.md#rule-2-when-an-implementation-of-an-already-shipped-public-api-must-call-asynchronous-code-and-block-for-its-completion-it-must-do-so-by-following-this-simple-pattern).

Reply:
Oh yeah, nice catch, we should beat whoever wrote that with a large 🐟

Reply:
*whistles*